coverage/README.md

```diff
@@ -0,0 +1,191 @@
+# Code Coverage Generation
+
+We have nightly code coverage generation so that we can track test coverage
+for Node.js, make the information public and then use that information
+to improve coverage over time.
+
+At this time we only capture coverage results once a day on linux x86. We 
+believe that coverage will not vary greatly across platforms and that the 
+process will be too expensive to run on every commit.  We will re-evaluate 
+these assumptions based on data once the process has been in place for 
+a while.
+
+This doc captures the infrastructure in place to support the generation
+of the coverage information published to coverage.nodejs.org.
+
+# Steps
+
+Generation/publication of the code coverage results consists of the following:
+
+* Nightly scheduled job - We have a job in jenkins which is scheduled to run at
+  11 EST each night. 
+  [node-test-commit-linux-coverage](https://ci.nodejs.org/view/All/job/node-test-commit-linux-coverage/).
+* At the end of the scheduled job it rsnyc's the generated data to the
+  benchmarking data machine.  We do this so that once the job is complete
+  the data is in a place where we know we can pull it from, and that pulling
+  that data will not affect any other jobs (for example jobs measuring
+  performance on the benchmark machine).
+* At hourly intervals the the data is rsync'd from the benchmarking
+  data machine to the website.  This is triggered from the nodejs.org website
+  machine and data is pulled from the benchmarking data machine. This allows
+  us to minimize who can modify the nodejs.org website as no additional 
+  access is required.
+
+# Benchmark Job
+
+The benchmark job follows the same pattern as our other build jobs in order
+to check out the version of node to be build/tested. It requires the following
+additions:
+
+
+1) Checkout of the scripts used to generate the coverage
+These will be moved to github./com/testing/coverage and the job
+updated once that is complete:
+```
+if [ ! -d node-core-coverage ]; then
+  git clone --depth=10 --single-branch git://github.com/addaleax/node-core-coverage.git
+fi
+```
+
+2) get a copy of gcov
+
+```
+# get gcov if required and then apply patches that are required for it
+# to work with node.js.  
+if [ ! -d gcovr ]; then
+  git clone --depth=10 --single-branch git://github.com/gcovr/gcovr.git
+  (cd gcovr && patch -p1 < "../node-core-coverage/gcovr-patches.diff")
+fi
+```
+
+3) install npms that we use to instrument node.js and generate javascript
```

Comment:
install -> Install
node.js -> Node.js (throughout the doc)
javascript -> JavaScript (throughout the doc)